parasolr/tests/test_query.py

```diff
@@ -117,6 +172,22 @@ def test_filter(self):
         assert 'date:1500' in filtered_qs.filter_qs
         assert 'name:he*' in filtered_qs.filter_qs
 
+    def test_facet(self):
+        mocksolr = Mock(spec=SolrClient)
+        sqs = SolrQuerySet(mocksolr)
+        # facet a search
+        facet_list = ['person_type', 'item_type']
+        faceted_qs = sqs.facet(*facet_list)
+        # faceting should be set on
+        assert faceted_qs.facet_opts['facet']
+        assert faceted_qs.facet_opts['facet.field'] == facet_list
+        # subsequents calls to facet should simply reset list
+        facet_list = ['foobars']
+        faceted_qs = faceted_qs.facet(*facet_list)
+        assert faceted_qs.facet_opts['facet']
+        assert faceted_qs.facet_opts['facet.field'] == facet_list
+
```

Comment:
Important case you're missing here - facet options should be preserved across other methods that return a new queryset, and should not modify the original. You're kind of missing it because you're testing that new facets replace previous facets. (Not sure what that should do for any previous facet options... do they reset too?)

I expect this to fail because I don't see any logic in the `_clone` method to copy facet options. Might be a good opportunity to write the failing test first.